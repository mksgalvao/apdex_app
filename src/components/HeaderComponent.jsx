import { React } from "react";

export const HeaderComponent = (props) => {
  return (
    <div>
      <h3>Apps by Host</h3>
      <p> For user: averylongemail@email.com</p>
    </div>
  );
};
