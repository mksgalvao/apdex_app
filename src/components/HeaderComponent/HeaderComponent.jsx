import { React } from "react";

import "./header.scss";

export const HeaderComponent = (props) => {
  const onInputSelect = () => {
    props.setVisualization();
  };
  return (
    <div className="headerWrapper ">
      <div className="textWrapper ">
        <h3 className="headerTitle   mr-1 ">Apps by Host</h3>
        <p className="email d-inline-block texte-left text-truncate">
          for user: averylongemailaddress@companyname.com
        </p>
      </div>
      <div className="inputWrapper d-flex align-items-center ">
        <input
          className="mb-1 mr-2"
          type="checkbox"
          id="visualization"
          onClick={onInputSelect}
        />
        <label className="label" htmlFor="visualization">
          {props.showList ? "Show as an awesome grid" : "Show as a list"}
        </label>
      </div>
    </div>
  );
};
