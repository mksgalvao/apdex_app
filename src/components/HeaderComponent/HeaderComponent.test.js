import React from "react";
import enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });

import { shallow } from "enzyme";
import { HeaderComponent } from "./HeaderComponent";

describe("Testing HeaderComponent", () => {
  let headerComponent;
  const setShowListMock = jest.fn();

  beforeEach(() => {
    headerComponent = shallow(
      <HeaderComponent showList={true} setVisualization={setShowListMock} />
    );
  });

  it("HeaderComponent should render", () => {
    expect(headerComponent).toMatchSnapshot();
  }),
    it("Should call setShowList when input checkbox was clicked", () => {
      const checkbox = headerComponent.find("input").at(0);
      checkbox.simulate("click", {});
      expect(setShowListMock).toHaveBeenCalledTimes(1);
    });
});
