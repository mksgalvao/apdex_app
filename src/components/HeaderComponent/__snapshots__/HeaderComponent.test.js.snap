// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HeaderComponent HeaderComponent should render 1`] = `ShallowWrapper {}`;
